src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionOptions.cs

```diff
@@ -28,7 +29,13 @@ public class HttpConnectionOptions
         public HttpConnectionOptions()
         {
             _headers = new Dictionary<string, string>();
-            _clientCertificates = new X509CertificateCollection();
+
+            // System.Security.Cryptography isn't supported on WASM currently
+            if (!Utils.IsRunningInBrowser())
```

Comment:
If you wanted to be extra nice, you could decorate ClientCertificates with `[UnsupportedOSPlatform("browser")]`

Reply:
I'm not familiar with that. Does it tell an analyzer that you shouldn't use that property? Will it complain at us using it internally for null checks?

Reply:
It'll warn if you use that property in a blazor application. Right now SignalR client doesn't claim to support browser so you wouldn't see the error internally. But it would print a warning in the null check once if you say the client targets browser.

Reply:
> SignalR client doesn't claim to support browser

Is this something we should look into in 6.0?

Reply:
It doesn't look like we have `[SupportedOSPlatform("browser")]` anywhere yet in this repo or in the runtime repo. I see `[UnsupportedOSPlatform("browser")]` in a couple places in both repos though.

Why would SignalR need to explicitly claim to support the "browser" platform? If it's opt-in, why put `[UnsupportedOSPlatform("browser")]` all over?